src/NuGet.Core/NuGet.Packaging/Rules/DependeciesGroupsForEachTFMRule.cs

```diff
@@ -0,0 +1,191 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using NuGet.Client;
+using NuGet.Common;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+using NuGet.RuntimeModel;
+
+namespace NuGet.Packaging.Rules
+{
+    class DependeciesGroupsForEachTFMRule : IPackageRule
+    {
+        HashSet<NuGetFramework> _compatNotExactMatches = new HashSet<NuGetFramework>();
+        HashSet<NuGetFramework> _noExactMatchesFromFile = new HashSet<NuGetFramework>();
+        HashSet<NuGetFramework> _noExactMatchesFromNuspec = new HashSet<NuGetFramework>();
+
+        public string MessageFormat { get; }
+
+        public string SecondWarning { get; }
+
+        public DependeciesGroupsForEachTFMRule(string messageFormat, string secondWarning)
+        {
+            MessageFormat = messageFormat;
+            SecondWarning = secondWarning;
+        }
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
+        {
+            var files = builder.GetFiles().
+                Where(t => PathUtility.GetPathWithDirectorySeparator(t).Count(m => m == Path.DirectorySeparatorChar) > 1);
+            var packageNuspec = builder.GetNuspec();
+            return Validate(files, packageNuspec);
+        }
+
+        internal IEnumerable<PackagingLogMessage> Validate(IEnumerable<string> files, Stream packageNuspec)
+        {
+            var managedCodeConventions = new ManagedCodeConventions(new RuntimeGraph());
+            var isCompatible = managedCodeConventions.Properties["tfm"].CompatibilityTest;
+            var collection = new ContentItemCollection();
+            collection.Load(files);
+
+            var libItems = GetContentForPattern(collection, managedCodeConventions.Patterns.CompileLibAssemblies);
+            var refItems = GetContentForPattern(collection, managedCodeConventions.Patterns.CompileRefAssemblies);
+            var libFrameworks = GetGroupFrameworks(libItems).ToArray();
+            var refFrameworks = GetGroupFrameworks(refItems).ToArray();
+
+            var tfmsFromFilesSet = new HashSet<NuGetFramework>();
+            tfmsFromFilesSet.AddRange(libFrameworks);
+            tfmsFromFilesSet.AddRange(refFrameworks);
+            var tfmsFromFiles = tfmsFromFilesSet.Where(t => t.Version.Major > 0).ToList();
```

Comment:
Can you explain why?

Reply:
When we have a file directly under the folder, it's returned as `.NETFramework0.0` so this alleviates that. In the original function, those files are weeded out it they only have 1 directory path separator

Reply:
Can you verify that it's .NET Framework 0.0 explicitly?

Reply:
So special case it?

Reply:
```cs
tfmsFromFilesSet.Where(t => t.Version.Major > 0)
```

I mean you are already special casing it :)
I'm recommending you be more explicit.